diff --git a/src/Assets.js b/src/Assets.js
--- a/src/Assets.js
+++ b/src/Assets.js
@@ -18,1 +18,2 @@
 var _imageSize = _interopRequireDefault(require("image-size"));
+var _imageSizeFromFile = require("image-size/fromFile");
@@ -173,1 +174,5 @@
-  const dimensions = isImage ? (0, _imageSize.default)(isImageInput) : null;
+  const dimensions = isImage
+    ? typeof isImageInput === "string"
+      ? await (0, _imageSizeFromFile.imageSizeFromFile)(isImageInput)
+      : (0, _imageSize.default)(isImageInput)
+    : null;
diff --git a/src/Assets.js.flow b/src/Assets.js.flow
--- a/src/Assets.js.flow
+++ b/src/Assets.js.flow
@@ -18,1 +18,2 @@
 import getImageSize from 'image-size';
+import {imageSizeFromFile} from 'image-size/fromFile';
@@ -223,1 +223,5 @@
-  const dimensions = isImage ? getImageSize(isImageInput) : null;
+  const dimensions = isImage
+    ? typeof isImageInput === 'string'
+      ? await imageSizeFromFile(isImageInput)
+      : getImageSize(isImageInput)
+    : null;